libs/labelbox/src/labelbox/schema/workflow/filter_utils.py

```diff
@@ -0,0 +1,775 @@
+"""Utilities for filter conversion in Project Workflows."""
+
+from typing import Dict, Any
+
+# Mapping of user-friendly names to API backend names
+ALIAS_TO_BACKEND_KEY = {
+    "created_by": "CreatedBy",
+    "labeled_by": "CreatedBy",
+    "annotation": "Annotation",
+    "labeled_at": "LabeledAt",
+    "sample_probability": "Sample",
+    "sample": "Sample",
+    "consensus_average": "ConsensusAverage",
+    "feature_consensus": "FeatureConsensusAverage",
+    "feature_consensus_average": "FeatureConsensusAverage",
+    "dataset": "Dataset",
+    "issue_category": "IssueCategory",
+    "batch": "Batch",
+    "metadata": "Metadata",
+    "model_prediction": "ModelPrediction",
+    "labeling_time": "LabelingTime",
+    "review_time": "ReviewTime",
+    "natural_language": "NlSearch",
+    "nl_search": "NlSearch",
+    "label_feedback": "LabelFeedback",  # If supported
+}
+
+# Reverse mapping from backend names to user-friendly names (prefer primary aliases)
+BACKEND_KEY_TO_ALIAS = {
+    "CreatedBy": "created_by",
+    "Annotation": "annotation",
+    "LabeledAt": "labeled_at",
+    "Sample": "sample",
+    "ConsensusAverage": "consensus_average",
+    "FeatureConsensusAverage": "feature_consensus_average",
+    "Dataset": "dataset",
+    "IssueCategory": "issue_category",
+    "Batch": "batch",
+    "Metadata": "metadata",
+    "ModelPrediction": "model_prediction",
+    "LabelingTime": "labeling_time",
+    "ReviewTime": "review_time",
+    "NlSearch": "natural_language",
+    "LabelFeedback": "label_feedback",
+}
```

Comment:
You should be able to create this mapping from `ALIAS_TO_BACKEND_KEY` to avoid duplication.

Reply:
Code refactored